lib/utils.js

```diff
@@ -30,9 +31,19 @@ const joinIterable = (val, sep) => {
   return res;
 };
 
+const checkTypeOrQuery = (value, name, type) => {
+  if (!(value instanceof QueryBuilder) && typeof value !== type) {
```

Comment:
It seems to me we need `OR` logic here: `!instance || !type`

Reply:
No, this is "exit condition" for error cases.